app/controllers/two_factor_authentication/otp_verification_controller.rb

```diff
@@ -24,7 +24,10 @@ def create
         increment_mfa_selection_attempt_count(otp_auth_method)
       end
       result = otp_verification_form.submit
-      post_analytics(result)
+
+      if UserSessionContext.confirmation_context?(context)
+        post_analytics(result)
```

Comment:
Optional: This method name could be improved to add a qualifier describing that it's specific to setup / confirmation. Right now it's confusing because we log analytics for both authentication and confirmation, but the method here only handles confirmation.

Reply:
Agreed. Renamed it in a1072ae to `log_confirmation_analytics`